chain/network/src/peer_manager.rs

```diff
@@ -169,13 +294,28 @@ impl PeerManagerActor {
         self.active_peers.insert(
             full_peer_info.peer_info.id,
             ActivePeer {
-                addr,
+                addr: addr.clone(),
                 full_peer_info,
                 sent_bytes_per_sec: 0,
                 received_bytes_per_sec: 0,
                 last_time_peer_requested: Utc.timestamp(0, 0),
             },
         );
+
+        // Sync routes to known account ids with new peer.
+        for mut announcement in
+            self.routing_table.account_peers.values_mut().map(|entry| entry.next_best_route())
+        {
+            // Update route with our information in case we are not the source.
+            if announcement.header().peer_id != self.peer_id {
```

Comment:
The header peer_id is irrelevant here, what matters is the `announcement.peer_id()` (i.e. the last peer, not the first peer).
I fixed it in chunks, but I guess it got lost during the merge.